rpc/src/rpc_impl.rs

```diff
@@ -421,8 +425,14 @@ impl<S: Storage + Send + core::marker::Sync + 'static> RpcFunctions for RpcImpl<
             .map(|(height, members)| PotentialFork { height, members })
             .collect();
 
+        let node_count = if network_metrics.node_count == 0 {
+            known_network.nodes().len()
+        } else {
+            network_metrics.node_count
```

Comment:
These should have the same value, since the `nodes` collection is built from the connections in both cases?

Reply:
not always; until the node gets any lists of peers, it has no `connections`, but it's very likely to have some `nodes` due to the calls to `update_height` caused by post-handshake pings.